drivers/rilmodem/sim.c

```diff
@@ -943,7 +944,8 @@ static void ril_query_pin_retries(struct ofono_sim *sim,
 			g_free(cbd);
 			CALLBACK_WITH_FAILURE(cb, NULL, data);
 		}
-	} else if (sd->vendor == OFONO_RIL_VENDOR_MTK) {
+	} else if (sd->vendor == OFONO_RIL_VENDOR_MTK ||
+			sd->vendor == OFONO_RIL_VENDOR_MTK2) {
```

Comment:
More than refactoring, maybe this should be called model/family/type or something like that instead of "vendor". Note also that in "enum ofono_vendor", defined by the atmodem driver we have:

	OFONO_VENDOR_GOBI,    (which is a qualcomm modem)
	OFONO_VENDOR_QUALCOMM_MSM,
	OFONO_VENDOR_WAVECOM,
	OFONO_VENDOR_WAVECOM_Q2XXX,
	OFONO_VENDOR_SIMCOM,
	OFONO_VENDOR_SIMCOM_SIM900,

so if changed, we should do that for this enum too.